wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/KubernetesToolApplier.java

```diff
@@ -135,6 +144,48 @@ private String retrieveContent(
         .collect(toList());
   }
 
+  /**
+   * Set {@link MACHINE_NAME_ATTRIBUTE} to commands which are configured in the specified tool.
+   *
+   * <p>Machine name will be set only if the specified recipe objects has the only one container.
+   */
+  private void estimateCommandsMachineName(
+      Devfile devfile, Tool tool, KubernetesList recipeObjects) {
+    List<Command> toolsCommands =
+        devfile
+            .getCommands()
+            .stream()
+            .filter(c -> c.getActions().get(0).getTool().equals(tool.getName()))
+            .collect(toList());
+    if (toolsCommands.isEmpty()) {
+      return;
+    }
+    List<Pod> pods =
+        recipeObjects
+            .getItems()
+            .stream()
+            .filter(hasMetadata -> hasMetadata instanceof Pod)
+            .map(hasMetadata -> (Pod) hasMetadata)
+            .collect(toList());
+
+    if (pods.size() != 1) {
+      // can not estimate commands machine name
+      return;
+    }
+
+    Pod pod = pods.get(0);
+    List<Container> containers = pod.getSpec().getContainers();
+
+    if (containers.size() != 1) {
+      // can not estimate commands machine name
```

Comment:
Could you describe why, not what?

Reply:
Actually would be better to combine both checks because the reason os probably the same